packages/gatsby/src/redux/run-sift.js

```diff
@@ -247,16 +201,20 @@ const applyFilters = (
   resolvedFields
 ) => {
   const filter = filterFields
-    ? getFilters(
-        liftResolvedFields(
-          toDottedFields(prepareQueryArgs(filterFields)),
-          resolvedFields
-        )
+    ? prefixResolvedFields(
+        createDbQueriesFromObject(prepareQueryArgs(filterFields)),
+        resolvedFields
       )
     : []
 
   let result
-  if (typedKeyValueIndexes && filter.length === 1) {
+  // Filters are flattened, so single filter will always be a chain with needle
+  // unless it's an elemmatch
+  if (
+    typedKeyValueIndexes &&
+    filter.length === 1 &&
+    filter[0].type !== `elemMatch`
```

Comment:
Was wondering whether this exception is still necessary?

Reply:
Why not?

Reply:
Because the filter gets resolved now. But maybe not resolved enough. I guess if you don't know the answer is "yes" :)

Reply:
They are resolved, but elemMatch is still a valid filter type. In future we might handle indexing for it too.